File: less1/F/py/main.py
from sys import stdin, stdout


def main():
    n = int(stdin.readline().strip())

    odd_count_is_even = True
    first_odd_pos = -1

    a = map(int, stdin.readline().strip().split())

    for i, v in enumerate(a):
        if v&1:
            odd_count_is_even = not odd_count_is_even
            if first_odd_pos == -1:
                first_odd_pos = i

    i = 0
    if odd_count_is_even:
        for _ in range(first_odd_pos):
            stdout.write("+")
        stdout.write("x")
        i = first_odd_pos + 1

    for _ in range(i, n-1):
        stdout.write("+")


if __name__ == "__main__":
    main()

File: less1/F/main.go
package main

import (
	"bufio"
	"io"
	"log"
	"os"
	"strconv"
)

func run(in io.Reader, out io.Writer) error {

	sc := bufio.NewScanner(in)
	sc.Split(bufio.ScanWords)

	sc.Scan()
	n, err := strconv.Atoi(sc.Text())
	if err != nil {
		return err
	}

	oddCountIsEven := true
	firstOddPos := -1

	for i := 0; i < n; i++ {

		sc.Scan()
		a, err := strconv.Atoi(sc.Text())
		if err != nil {
			return err
		}

		if a&1 == 1 {
			oddCountIsEven = !oddCountIsEven 
			if firstOddPos == -1 {
				firstOddPos = i
			}
		}
	}

	w := bufio.NewWriter(out)

	i := 1
	if oddCountIsEven {
		for n := firstOddPos+1; i < n; i++ {
			w.WriteByte('+')
		}
		w.WriteByte('x')
		i++
	}

	for ; i < n; i++ {
		w.WriteByte('+')
	}

	w.WriteByte('\n')
	w.Flush()

	return nil
}

func main() {
	err := run(os.Stdin, os.Stdout)
	if err != nil {
		log.Fatal(err)
	}
}
